core/src/main/java/tech/pegasys/eth2signer/core/http/SigningRequestHandler.java

```diff
@@ -61,14 +62,14 @@ private void generateResponseFromBody(
           .end("Request body illegally formatted for signing operation.");
       return;
     }
-    final Optional<ArtifactSigner> signer = signerProvider.getSigner(signingRequest.publicKey());
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner(publicKey);
```

Comment:
nit: you _could_ move this up into the "handle" function - i.e. this function doesn't need to know about the public key, just a signer.